src/main/java/org/primefaces/component/splitbutton/SplitButtonRenderer.java

```diff
@@ -157,16 +159,19 @@ protected void encodeMenuIcon(FacesContext context, SplitButton button, String i
             writer.writeAttribute("disabled", "disabled", "disabled");
         }
 
-        //icon    
-        writer.startElement("span", null);
-        writer.writeAttribute("class", "ui-button-icon-left ui-icon ui-icon-triangle-1-s", null);
-        writer.endElement("span");
+        
+        if (button.getChildCount() > 0) { 
```

Comment:
@jxmai Was this test really necessary? Since `encodeMenuIcon` is called by `encodeMarkup``where the test is made

Reply:
Alright, I see. This if statement seems redundant because of https://github.com/primefaces/primefaces/blob/master/src/main/java/org/primefaces/component/splitbutton/SplitButtonRenderer.java#L90. but since you have raised another PR https://github.com/primefaces/primefaces/pull/3503 with better enhancement to deal with the same ticket, I will leave the task up to you :smile: